docs/en/stack/ml/df-analytics/dfa-regression.asciidoc

```diff
@@ -0,0 +1,103 @@
+  apartment either has a riverside view or doesn't have one.
+Arrays are not supported.
+
+
+[discrete]
+[[dfa-regression-supervised]]
+=== Training the {regression} algorithm
+
+{regression-cap} is a supervised machine learning process, which means that you 
+need to supply a labeled training dataset that has some {feature-vars} and a 
+{depvar}. The {regression} algorithm learns the relationships between the 
+features and the {depvar}. Once you've trained the model on your training 
+dataset, you can reuse the knowledge that the model has learned about the 
+relationships between the data points to make certain inferences about new data 
+points.
+
+The relationships between the {feature-vars} and the {depvar} is described as a 
+mathematical formula. {reganalysis-cap} tries to find the best value for the 
+{depvar} by combining various {ml} techniques, that means that in the {stack}, 
+we use an ensemble learning model that uses multiple algorithms. The performance 
+of an ensemble is usually better than the performance of an individual 
+algorithm because each of the algorithms has a different bias, so each of them 
+falls into different traps. When they are used together as a group, then they 
+capture the characteristics of the data better. The ensemble learning technique 
+that we use in the {stack} is a type of boosting called extreme gradient boost 
+(XGboost) which combines decision trees with gradient boosting methodologies.
+
+ 
+[discrete]
+[[dfa-regression-evaluation]]
+=== Measuring model performance
+
+You can measure how well the model has performed on your training dataset by 
+using the `regression` evaluation type of the 
+{ref}/evaluate-dfanalytics.html[evaluate {dfanalytics} API]. The mean squared 
+error (MSE) value that the evaluation provides you on the training dataset is 
+the _training error_. Training the {regression} model means finding the 
+combination of model parameters that produces the lowest possible training 
+error.
+
+Another crucial measurement is how well your model performs on unseen 
+data points. To assess how well the trained model will perform on data it has 
+never seen before, you must set aside a proportion of the training dataset for 
+testing. This split of the dataset is the testing dataset. Once the model has 
+been trained and found out the required coefficients, you can let the model 
+predict the value of the data points it has never seen before and compare the 
+prediction to the actual value. This test provides an estimate of a quantity 
+known as the _model generalization error_.
+
+Two concepts describe how well the {regression} algorithm was able to learn the 
+relationship between the {feature-vars} and the {depvar}. _Underfitting_ is when 
+the model cannot capture the complexity of the dataset. _Overfitting_ is when 
+the model has trained so specifically to a dataset that it is capturing all the 
+details and cannot generalize well enough. A model that overfits the data has a 
```

Comment:
I'd rephrase this slightly. The details could be things which do generalise. The quantities which don't generalise are things which are unique to the training data and usually called noise. How about:

_Overfitting_ is when the model is too specific to the training data set and is capturing details which do not generalize to new data.